### DISCUSSION TITLE
Load Testing and Multiple users

### DISCUSSION THREAD

**Discussion by didierrc** [Arquisoft/faq#64] [Q&A]
Hi everyone 👋,

My team wiq_en3b have successfully implemented the load testing with Gatling and some of them are working pretty well. 

The problem arises with the Recorder tool since seems a bit "limited". For instance, we wanted to test the following scenario: **Different players login into our Application, starting a game and answering to questions** to see how many users can our Application handle concurrently. 

The thing is that if we use the Recorder and we login with some credential (e.g. "pepe" and "loadPass123"), the Java test generated by it always input the same credentials, so the simulation will be: same user playing multiple games during a period of time.

Is there any solution to it 😓? I have tried to look and modify the Java code but all my attempts to change this behaviour were unsuccessful. I am uploading the starting code of a gameplay in which the user only access to the game (without answering). This code is commented so you can get the "vibe" of what is going on with my tests.
```java
public class GameSimulation extends Simulation {

  // httpProtocol: the protocol captured by Recorder
  // headers_N: Series of headers captured by Recorder
 
  private String uri1 = "localhost";

  // Request_0: Enters into http://localhost/ (main page of game - shows up the LogIn)
  // Request_0-5: Bunch of resources like js,css or translation assets.
  // Request_6-7: Sending (POST) the credentials of the user to enter the game.
  // kaw/gamesimulation/007_request.json has the following structure: {username: "pepe", password: "somePassword123"} 
  // Request_8: Starting a new game (GET questions)

  private ScenarioBuilder scn = scenario("GameSimulation")
    .exec(
      http("request_0")
        .get("http://" + uri1 + "/")
        .headers(headers_0)
        .resources(
          http("request_1")
            .get("http://" + uri1 + "/static/js/main.9dbc8138.js")
            .headers(headers_1),
          http("request_2")
            .get("http://" + uri1 + "/static/css/main.4fe7b79e.css")
            .headers(headers_2),
          http("request_3")
            .get("http://" + uri1 + "/KaW.png")
            .headers(headers_3),
          http("request_4")
            .get("http://" + uri1 + "/assets/locales/es/translation.json")
            .headers(headers_4),
          http("request_5")
            .get("http://" + uri1 + "/assets/locales/en/translation.json")
            .headers(headers_5)
        ),
      pause(8),
      http("request_6")
        .options("/login")
        .headers(headers_6)
        .resources(
          http("request_7")
            .post("/login")
            .headers(headers_7)
            .body(RawFileBody("kaw/gamesimulation/0007_request.json")) // here always inject the same user!!
        ),
      pause(12),
      http("request_8")
        .get("/questions?size=5&lang=en")
        .headers(headers_8)
    );

  {
	  setUp(scn.injectOpen(constantUsersPerSec(50).during(60).randomized())).protocols(httpProtocol);
  }
}
```
I am tagging @pglez82 since he asked me to publish my question here!! Have a nice week :)

**Comment by pglez82** [comment]
You are right that the recorder is not enough for this kind of scenarios in which we want to simulate different users working in our app. There are several techniques to achieve this in Gatling. I refer you to an introductory article about this topic (https://gatling.io/blog/gatling-credentials)

In particular I think you need a Feeder. In this link (https://docs.gatling.io/reference/script/core/session/feeders/), you have information and examples on how to load users from a CSV for instance. Obviously these users need to exist in your app beforehand, so you may need to populate your database before launching the load tests.

Let me know if it helps or you need more help.
